src/main/java/com/avairebot/imagegen/RankBackgroundHandler.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2018.
+ *
+ * This file is part of AvaIre.
+ *
+ * AvaIre is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * AvaIre is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AvaIre.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ */
+
+package com.avairebot.imagegen;
+
+import com.avairebot.contracts.imagegen.BackgroundRankColors;
+import com.avairebot.shared.ExitCodes;
+import com.avairebot.utilities.ResourceLoaderUtil;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.net.JarURLConnection;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.*;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+public class RankBackgroundHandler
+{
+    private static final Logger log = LoggerFactory.getLogger(RankBackgroundHandler.class);
+    public static final RankBackground DEFAULT_BACKGROUND = new RankBackground(2, 10, "Purple", null, new BackgroundRankColors());
```

Comment:
The default background should be removed since there is no guarantee that the background actually exists.

Reply:
What do you mean there is no guarantee it actually exists? The default rank colors return the colors needed for the purple background, though I can see why it should be probably be removed. I will get it uncoupled to the Renderer then.

Reply:
As described in the top post to the PR review, all the rank backgrounds should basically be loaded from the `backgrounds` directory, not the one under `resources`, but the one that is created when the bot starts up, which means the first time the bot is starting up and is creating the `backgrounds` directory, it should basically copy all the contents of the folder in the `resources` directory so the folder has all the contents, configs and images included, so self-hosters can see examples of existing backgrounds, as-well-as edit or delete the backgrounds that come with the bot.